dns_sd_windows.c

```diff
@@ -0,0 +1,262 @@
+// Based on the example provided here: https://github.com/mjansson/mdns
+
+#ifdef _WIN32
```

Comment:
If the CMake is defining things as only Windows, can we remove some of these #ifdef _WIN32? This file will only be compiled on Windows - it just makes is harder to follow.

Reply:
If the plan is to remove avahi and bonjour, and only use this - then that is different.... (Since we can't remove avahi - we use it in `iiod` to advertise - I see this being a Windows only file for a long time.